### PR TITLE
Port: Added a chester port, a very simple GameBoy emulator.

### PR DESCRIPTION
**Pull request by Baitinq**
Hey friends,

Ive been a long time lurker and finally got the energy to contribute to this amazing project. Found a pretty cool portable GameBoy emulator and decided to try and get it to work on serenity. (https://github.com/veikkos/chester)

Its only dependency is SDL2. Doesnt support gameboy advanced games.

I believe it works and builds fine but there might be some mistakes as it is my first port. :^)

### CODE REVIEW

**Comment by Baitinq** [comment]
EDIT: Added the chester port to the AvaileablePorts.md list.

**Review by linusg — CHANGES_REQUESTED** [review]
Please don't use your personal fork as the source for a port - ports should either use the upstream repository, with any required patches living within the serenity repo, or a fork of the project in the SerenityOS organization.
The repo used here (https://github.com/Baitinq/Chester_SerenityOS) has a single commit titled "XD", this doesn't seem like a good idea to me. It's also unclear what changes are required for serenity as the git history is gone.
TL;DR: please use upstream source and provide required changes as patch files.

**Comment by Baitinq** [comment]
@linusg Should i then edit the package script to apply the patches?

**Comment by linusg** [comment]
No, that happens automatically, just put them in a directory called patches inside the chester port directory. Look at some other ports to see how this works, it's quite common :)

**Comment by Baitinq** [comment]
@linusg thanks for your help, will do.

**Comment by Baitinq** [comment]
EDIT: According to @linusg suggestions the port now downloads the src from the original source and the patches are applied separately :)

File: Ports/chester/package.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+port=chester
+useconfigure=true
+version=1.0
+depends="SDL2"
+workdir=chester-public
+configopts="-DCMAKE_TOOLCHAIN_FILE=$SERENITY_ROOT/Toolchain/CMakeToolchain.txt"
+files="https://github.com/veikkos/chester/archive/refs/heads/public.zip chester.zip"
+
+configure() {
+	run cmake $configopts
```
Comment (linusg) [inline Ports/chester/package.sh:11]:
```suggestion
    run cmake $configopts
```

File: Ports/chester/package.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+install() {
+    run cp bin/chester $SERENITY_ROOT/Build/i686/Root/bin
```
Comment (linusg) [inline Ports/chester/package.sh:15]:
```suggestion
    run cp bin/chester "${SERENITY_ROOT}/Build/${SERENITY_ARCH}/Root/bin"
```

**Review by linusg — COMMENTED** [review]
Nice!

File: Ports/chester/package.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash ../.port_include.sh
```
Comment (linusg) [inline Ports/chester/package.sh:1]:
```suggestion
 #!/usr/bin/env -S bash ../.port_include.sh
```

File: Ports/chester/package.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+workdir=chester-public
+configopts="-DCMAKE_TOOLCHAIN_FILE=$SERENITY_ROOT/Toolchain/CMakeToolchain.txt"
+files="https://github.com/veikkos/chester/archive/public.tar.gz chester.tar.gz
+
+configure() {
+    run cmake $configopts
+}
+
+install() {
+    run cp bin/chester "${SERENITY_ROOT}/Build/${SERENITY_ARCH}/Root/bin"
```
Comment (linusg) [inline Ports/chester/package.sh:15]: Sorry, I forgot there's a variable for this:

```suggestion
    run cp bin/chester "${SERENITY_BUILD_DIR}/Root/bin"
```

**Review by linusg — COMMENTED** [review]
two more small changes... please also squash the commits as per the [contibution guidelines](https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md#code-submission-policy) :)

**Comment by Baitinq** [comment]
@linusg My bad, I have now squashed both commits and applied both of your suggestions.

**Review by linusg — APPROVED** [review]
Awesome, LGTM! cc @danboid, you might like this :)
